src/Sum/README.md

```diff
@@ -0,0 +1,105 @@
+# Sum
```

Comment:
A majority of comments on `Prod` apply to `Sum` as well around throwing due to wrong type, spacing and comments.

Reply:
Also same as `Prod`, `Sum` misses same type of example

Reply:
I left `Sum` and `Prod` with nearly identical docs because they aren't very different, but if I can think of a good example for `Sum`, I'll be replacing it